server/src/main/java/org/elasticsearch/index/mapper/DataStreamTimestampFieldMapper.java

```diff
@@ -195,37 +195,25 @@ public void postParse(DocumentParserContext context) throws IOException {
             // not configured, so skip the validation
             return;
         }
-        boolean foundFsTimestampField = false;
-        IndexableField first = null;
-        final List<IndexableField> fields = context.rootDoc().getFields();
-        for (int i = 0; i < fields.size(); i++) {
-            IndexableField indexableField = fields.get(i);
-            if (DEFAULT_PATH.equals(indexableField.name()) == false) {
-                continue;
-            }
-            if (first == null) {
-                first = indexableField;
-            }
-            if (indexableField.fieldType().docValuesType() == DocValuesType.SORTED_NUMERIC) {
```

Comment:
I don't fully understand this check and how should it be replicated. As far as i can tell if `@timestamp` field does not have doc values, it will not pass the validation performed by this mapper and we should never see that?

Reply:
When multiple timestamps have been parsed by `DateFieldMapper` and the doc values have been enabled which is required for the `@timestamp` field, then there are multiple `IndexableField` instances (for points and doc values). This just counts the doc value ones and fails when a second one is encountered.

Reply:
I see, without this check `encountered multiple values` would be thrown even for a single value of `@timestamp`.

Reply:
Correct, because all IndexableField instances would count also points, which by default is always the case.